shared/serial_port_controller.py

```diff
@@ -96,10 +96,48 @@ def close_reader_port(self):
         if self.reader_port is not None:
             self.reader_port.close()
 
-    def read_info(self):
+    def open_reader_port(self, port_name):
```

Comment:
Properly using the saved settings from the .csv file to auto-populate serial settings. Good stuff.